src/arrays/ladderops.jl

```diff
@@ -22,10 +22,22 @@
     before_eye(lens, pivot) = speye(prod(lens[[1:pivot-1]]))
     after_eye(lens, pivot) = speye(prod(lens[[pivot+1:length(lens)]]))
     function eye_sandwich(lens, pivot, op)
-        return kron(before_eye(lens, pivot), 
-                    op, 
+        return kron(before_eye(lens, pivot),
+                    op,
                     after_eye(lens, pivot))
     end
 
+function positionop(n::Int)
+    cop = raiseop(n)
+    return (cop+cop')/sqrt(2.)
+end
+
+function momentumop(n::Int)
+    cop = raiseop(n)
+    return im*(cop-cop')/sqrt(2.)
+end
+
```

Comment:
@jrevels if I my understanding of the first comment is right  it meant that we should be able to track intermediate steps in producing the final answer. In specific have functions which produce the intermediate results and reuse them as necessary ?? But in these case it is a straight forward implementation. Please do correct me if I am missing something.